### DISCUSSION TITLE
libcrashlytics-trampoline does not have a stack canary value added to the stack

### DISCUSSION THREAD

**Discussion by sabrina-tw** [invertase/react-native-firebase#6819] [General]
We have recently run some security scanning and found that libcrashlytics-trampoline does not have a stack canary value added to the stack:

- lib/arm64-v8a/libcrashlytics-trampoline.so
- lib/armeabi-v7a/libcrashlytics-trampoline.so
- lib/x86_64/libcrashlytics-trampoline.so

Stack canaries are used to detect and prevent exploits from overwriting return address.
It can be enabled by using the option -fstack-protector-all.

Could we look into enabling stack canaries for libcrashlytics-trampoline?

Thanks!

**Comment by SiddheshShetye** [comment]
will this be fixed?
